app/src/main/cpp/skyline/soc/gm20b/engines/maxwell_dma.cpp

```diff
@@ -30,167 +31,241 @@ namespace skyline::soc::gm20b::engine {
     }
 
     void MaxwellDma::LaunchDma() {
-        if (*registers.lineLengthIn == 0)
-            return; // Nothing to copy
+        DmaCopy();
 
-        if (registers.launchDma->remapEnable) {
-            Logger::Warn("DMA remapping is unimplemented!");
-            return;
-        }
+        ReleaseSemaphore();
+    }
 
+    void MaxwellDma::DmaCopy() {
         if (registers.launchDma->multiLineEnable) {
+            if (registers.launchDma->remapEnable) [[unlikely]] {
+                Logger::Warn("Remapped DMA copies are unimplemented!");
+                return;
+            }
+
             channelCtx.executor.Submit();
-            if (registers.launchDma->srcMemoryLayout == Registers::LaunchDma::MemoryLayout::Pitch &&
-                registers.launchDma->dstMemoryLayout == Registers::LaunchDma::MemoryLayout::BlockLinear)
-                CopyPitchToBlockLinear();
-            else if (registers.launchDma->srcMemoryLayout == Registers::LaunchDma::MemoryLayout::BlockLinear &&
-                registers.launchDma->dstMemoryLayout == Registers::LaunchDma::MemoryLayout::Pitch)
+
+            if (registers.launchDma->srcMemoryLayout == registers.launchDma->dstMemoryLayout) [[unlikely]] {
```

Comment:
Tbh i'd drop hints here

Reply:
Pitch to pitch/blocklinear to blocklinear copies are rare enough that i'd keep them personally